### PR TITLE
contirbuting a Joomla feature

### PR DESCRIPTION
**Pull request by jeroenbrons**
contirbuting Joomla support
i ran Ansible-lint

### CODE REVIEW

File: templates/traefik/traefik.toml [hunk 1]
```diff
@@ -163,7 +163,7 @@ email = "{{ ansible_nas_email }}"
 storage = "acme.json"
 entryPoint = "https"
 acmeLogging = true
-onDemand = false # create certificate when container is created
+onDemand = true # create certificate when container is created
```
Comment (davestephens) [inline templates/traefik/traefik.toml]: Can you revert this change please

File: templates/traefik/traefik.toml [hunk 2]
```diff
@@ -204,4 +204,6 @@ onDemand = false # create certificate when container is created
           "thelounge.{{ ansible_nas_domain }}",
           "transmission.{{ ansible_nas_domain }}",
           "transmission-openvpn.{{ ansible_nas_domain }}",
-          "znc.{{ ansible_nas_domain }}"]
+          "znc.{{ ansible_nas_domain }}",
+          "joomla.{{ ansible_nas_domain }}"
```
Comment (davestephens) [inline templates/traefik/traefik.toml]: You may (or may not) have noticed that the rest of the hostnames are alphabetical :)

File: Vagrantfile [hunk 3]
```diff
@@ -2,8 +2,8 @@ Vagrant.require_version ">= 2.2.2"
 
 Vagrant.configure(2) do |config|
   config.vm.box = "ubuntu/bionic64"
-  config.vm.network "private_network", ip: "172.30.1.5"
-  config.ssh.insert_key = false
+  config.vm.network "public_network", ip: "192.168.0.27", bridge: "Broadcom 802.11n Network Adapter"
+  config.ssh.insert_key = true
```
Comment (davestephens) [inline Vagrantfile]: Can you revert your changes to this file please, if these changes were for a good reason please can you submit a separate PR?

File: tasks/joomla.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  with_items:
+    - "{{ docker_home }}/joomla"
+
+- name: Create MySQL Directorys for Joomla
+  file:
+    path: "{{ item }}"
+    state: directory
+    # mode: 0755
+  with_items:
+    - "{{ docker_home }}/JoomlaDB"
```
Comment (davestephens) [inline tasks/joomla.yml]: Please can you combine this with the previous step - `with_items` allows you to provide a list.

Additionally, the DB and Joomla data should fall within the same subdirectory, ie:
```
{{ docker_home }}/joomla/data
{{ docker_home }}/joomla/data
```

File: tasks/joomla.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+- name: MySQL Docker Container for Joomla
+  docker_container:
+    name: mysql_joomla
+    image: mysql
```
Comment (davestephens) [inline tasks/joomla.yml]: What version of MySQL? If a user has watchtower enabled this will constantly pull the latest available which may be undesired. Perhaps better to specify an image version tag too.

**Comment by davestephens** [comment]
Thanks for your contribution! :+1: I've added a few comments that will need addressing before I'll be happy to merge this. 

Also one other big thing you're missing - docs!

File: tasks/joomla.yml (hunk 5 above)
Comment (jeroenbrons) [reply tasks/joomla.yml]: later on i did

File: Vagrantfile (hunk 3 above)
Comment (jeroenbrons) [reply Vagrantfile]: reverted

**Comment by davestephens** [comment]
Still missing docs :-( In the docs please can you cover that users should change the `changeme` username and password in `all.yml.dist`.

**Comment by jeroenbrons** [comment]
fially found time to fix

**Comment by davestephens** [comment]
This needed a bit of love (mostly around docs) - I've tidied it up and merged it :+1:
